app/v1/support/DownstreamResponseMappingSupport.scala

```diff
@@ -16,34 +16,44 @@
 
 package v1.support
 
+import play.api.libs.json.Writes
 import utils.Logging
 import v1.controllers.EndpointLogContext
 import v1.models.errors._
 import v1.models.outcomes.ResponseWrapper
+import play.api.libs.json.{JsObject, Json, Writes}
 
-trait DesResponseMappingSupport {
+trait DownstreamResponseMappingSupport {
   self: Logging =>
 
-  final def mapDesErrors[D](errorCodeMap: PartialFunction[String, MtdError])(desResponseWrapper: ResponseWrapper[DesError])(implicit
-      logContext: EndpointLogContext): ErrorWrapper = {
+  final def validateRetrieveResponse[T: Writes](downstreamResponseWrapper: ResponseWrapper[T]): Either[ErrorWrapper, ResponseWrapper[T]] = {
+    if (Json.toJson(downstreamResponseWrapper.responseData) == JsObject.empty) {
+      Left(ErrorWrapper(downstreamResponseWrapper.correlationId, NotFoundError, None))
+    } else {
+      Right(downstreamResponseWrapper)
```

Comment:
I think you've improved on the template here. Nice to see no desResponseWrappers!